### PR TITLE
Virtual kubelet: improve certificates management

### LINKED ISSUE #1334: kubectl run/exec fails with x509 error due to invalid hostname
#### What happened:

- Setup a provider and a consumer cluster in GKE, had to disable the PodCIDR validation check as it is quite normal in GKE for PodIPs to be different from the Cluster PodCIDRs. 
- Created and offloaded a namespace in the one cluster and offloaded it to the other cluster successfully.
- Ran a pod like below 
``` kubectl --context=cluster-1 -n offtest run nginx2 -it --image=docker.io/clearlinux/nginx:latest -- /bin/bash 
``` 
This failed with below error -
```
If you don't see a command prompt, try pressing enter.
Error attaching, falling back to logs: error dialing backend: x509: certificate is valid for 10.192.0.2, not 10.192.1.2
Error from server: Get "https://10.192.1.2:10250/containerLogs/offtest/nginx2/nginx2": x509: certificate is valid for 10.192.0.2, not 10.192.1.2
```
Where the 10.192.1.2 IP belongs to liqo-cluster-1 virtual node and none of the node IPs have the 10.192.0.2 IP for which the cert is provisioned. 

#### What you expected to happen:
Pod was created and attached to and I see a bash prompt.

#### How to reproduce it (as minimally and precisely as possible):
- Create two GKE clusters in a single GCP project with 2 separate VPCs containing overlapping IPV4 ranges.
- Setup Liqo, offload a namespace and try to run a pod in the offloaded namespace using kubectl exec
#### Anything else we need to know?:
When adding a Node Pool the Node Pool Pod Address Range might need to be overridden to be different from the Cluster's.

#### Environment:
- Liqo version: 0.5.2
- Kubernetes version (use `kubectl version`): v1.22.8-gke.202
- Cloud provider or hardware configuration: Google Cloud / GKE
- Network plugin and version: 
- Install tools: Custom compiled liqoctl to disable PodCIDR validation.
- Others:

### PR DESCRIPTION
**Pull request by giorio94**
# Description

This PR refactors the logic used by the virtual kubelet to generate the certificate of the server (i.e., for logs/exec/metrics), in order to ensure its correctness when the pod is restarted (thus, changing the internal node IP), as well as supporting renewal when appropriate.

Fixes #1334

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Existing tests
- [x] Manually, on GCP

### CODE REVIEW

**Review by aleoli — APPROVED** [review]

File: cmd/init-virtual-kubelet/main.go [hunk 1]
```diff
@@ -14,138 +14,5 @@
 
 package main
```
[the remark this answers is not in the document]
Comment (giorio94) [reply cmd/init-virtual-kubelet/main.go:15]: Definitely not. It was still there to make sure tests could complete correctly, due to the CI issue when removing a component. I'm going to remove that package and then "force" merge this PR.

**Pull request merged**